controllers/order.js

```diff
@@ -0,0 +1,99 @@
+const asyncHandler = require('express-async-handler');
+const Order        = require('../models/Order');
+
+
+// @route POST /orders/
+// @descp Create a new Order
+// @access Private
+exports.createOrder = asyncHandler(async (req , res , next)=>{
+    const newOrder = new Order(req.body);
+    const savedOrder = await newOrder.save();
+    if(!savedOrder){
+        res.status(500).json({error : "Something went wrong"})
+    }else{
+        res.status(200).json({order : savedOrder})
+    }
+})
+
+// @route GET /orders/:id
+// @descp Get an Order
+// @access Private
+exports.getAnOrder = asyncHandler(async (req , res , next)=>{
+    const orderId = req.params.id;
+    const foundOrder = await Order.findById(orderId);
+    if(!foundOrder){
+        res.status(500).json({error : "something went wrong"});
```

Comment:
I think it's 404 error in here
`res.status(404).json({error : "resource could not be found !"});`

Reply:
HTTP Status Codes > [ref ](https://restfulapi.net/http-status-codes/#:~:text=404%20(Not%20Found),condition%20is%20temporary%20or%20permanent.)